mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -151,4 +167,64 @@ class StringIndexerModel private[ml] (
     val copied = new StringIndexerModel(uid, labels)
     copyValues(copied, extra)
   }
+
+  /**
+   * Return a model to perform the inverse transformation.
+   * Note: by default we keep the original columns during this transformation
+   * so the invert should only be needed if you do something beyond simply
+   * applying the original transformation.
+   */
+  def invert(): StringIndexerInvertModel = {
+    new StringIndexerInvertModel(uid, labels)
```

Comment:
Another option would be to make StringIndexerInverseModel not be a Model, which actually might make more sense.